### PR TITLE
upgrade project to Phoenix 1.4

### PR DESCRIPTION
**Pull request by SimonLab**
ref: #14
- Update structure project to Phoenix 1.4 following Chris guide: https://gist.github.com/chrismccord/bb1f8b136f5a9e4abc0bfc07b832257e

- Use ` naive_datetime_usec` type to save microseconds of the timestamps (inserted_at and updated_at) in postgres.

- update Readme to fix small typos:
  - add sql command to define password for user
  - update ecto command variable in one of the example (query instead of sub)
  - update link for Behaviour page documentation (hexdoc return a 404)

### CODE REVIEW

File: assets/package-lock.json [hunk 1]
```diff
@@ -2,3724 +2,8627 @@
   "requires": true,
```
Comment (nelsonic) [inline assets/package-lock.json]: please add `assets/package-lock.json` to `.gitignore` it's a _super_ "_noisy_" file.

File: .gitignore [hunk 2]
```diff
@@ -29,4 +29,7 @@ npm-debug.log
 # variables.
 /config/*.secret.exs
 
-.elixir_ls
\ No newline at end of file
+.elixir_ls
+
+Dockerfile
```
Comment (nelsonic) [inline .gitignore]: what are you dockerising for this example?

Comment (SimonLab) [reply .gitignore]: I'm running Elixir Phoenix Elm with Docker:
```
FROM elixir:1.7.3

RUN mix local.hex --force \
  && mix archive.install hex phx_new 1.4.0 --force \
  && apt-get update \
  && curl -sL https://deb.nodesource.com/setup_8.x | bash \
  && apt-get install -y apt-utils \
  && apt-get install -y nodejs \
  && apt-get install -y build-essential \
  && apt-get install -y inotify-tools \
  && mix local.rebar --force \
  && wget "https://github.com/elm/compiler/releases/download/0.19.0/binaries-for-linux.tar.gz" \
  && tar xzf binaries-for-linux.tar.gz \
  && mv elm /usr/local/bin/

ENV APP_HOME /app
WORKDIR $APP_HOME


CMD ["mix", "phx.server"]
```

and I'm using docker-compose.yml to use Postgres:
```
version: '3'
services:
  app:
    build: .
    ports:
      - "4000:4000"
    volumes:
      - .:/app
    depends_on:
      - db
    env_file:
      - ./.env
  db:
    image: postgres:10.5
    ports:
        - "5432:5432"
```

My machine doesn't have Postgres installed anymore (I had some issues with psql) and I find it easier to run the application with Docker.
Howerver it's not directly linked to the project and I don't want to push these files to the repository.

Comment (nelsonic) [reply .gitignore]: @SimonLab makes sense. please add your setup to https://github.com/dwyl/learn-docker as discussed in https://github.com/dwyl/learn-docker/issues/15 thanks! ✨

File: README.md [hunk 3]
```diff
@@ -391,7 +408,7 @@ and **then** run:
 Now that we have no way to delete or update the data,
 we need to define the functions we'll use to access and insert the data.
 To do this we'll define an
-[Elixir behaviour](https://hexdocs.pm/elixir/behaviours.html)
+[Elixir behaviour](https://elixir-lang.org/getting-started/typespecs-and-behaviours.html#behaviours)
```
Comment (nelsonic) [inline README.md]: 👍

File: README.md [hunk 4]
```diff
@@ -702,8 +719,8 @@ defmodule Append.AddressTest do
     end
 
     test "all/0" do
-      {:ok, item} = insert_address()
-      {:ok, item_2} = insert_address("Loki")
+      {:ok, _item} = insert_address()
```
Comment (nelsonic) [inline README.md]: why does `_item` have a preceding underscore?

File: config/dev.exs [hunk 5]
```diff
@@ -56,7 +57,7 @@ config :phoenix, :stacktrace_depth, 20
 
 # Configure your database
 config :append, Append.Repo,
-  adapter: Ecto.Adapters.Postgres,
+  migration_timestamps: [type: :naive_datetime_usec],
```
Comment (nelsonic) [inline config/dev.exs]: 👍

**Pull request merged**